packages/terminal/src/browser/terminal-frontend-contribution.ts

```diff
@@ -185,10 +185,10 @@ export class TerminalFrontendContribution implements TerminalService, CommandCon
             this.onDidChangeCurrentTerminalEmitter.fire(this._currentTerminal);
         }
     }
-    protected updateCurrentTerminal(): void {
-        const widget = this.shell.currentWidget;
-        if (widget instanceof TerminalWidget) {
-            this.setCurrentTerminal(widget);
+    protected updateCurrentTerminal(widget?: TerminalWidget): void {
+        const currentWidget = widget || this.shell.currentWidget;
+        if (currentWidget instanceof TerminalWidget) {
+            this.setCurrentTerminal(currentWidget);
```

Comment:
i made this change because i found that
if we do 
```
const newWidget = widgetManager.getOrCreateWidget(ID, options);
this.applicationShell.revealWidget(newWidget.id);
```
`this.shell.currentWidget` is the most recent widget that I clicked, not the terminal widget I just created.

therefore, in my opinion, in the context of `TerminalFrontendContribution` class, I made the `currentTerminal` the one that we opened (i.e.., activate or reveal) most recently.

What do you think?
@vince-fugnitto @RomanNikitenko @akosyakov @lmcbout

Reply:
please revert, current is last focused, reveal does not change focus, you should use  `activate` if you want to give a focus to the terminal

Reply:
> this.shell.currentWidget is the most recent widget that I clicked, not the terminal widget I just created.

You should use `TerminalFrontendContribution.currenTerminal`.

> therefore, in my opinion, in the context of TerminalFrontendContribution class, I probably should make the currentTerminal the one that we opened (i.e.., activate or reveal) most recently.

Semantics of `TerminalFrontendContribution.currenTerminal` is correct and cannot be changed. It is a currently focused terminal or one which most recently had focused. Not anything else.